octodns/provider/azuredns.py

```diff
@@ -330,13 +330,31 @@ def __init__(self, id, client_id, key, directory_id, sub_id,
                        'key=***, directory_id:%s', id, client_id, directory_id)
         super(AzureProvider, self).__init__(id, *args, **kwargs)
 
-        credentials = ServicePrincipalCredentials(
-            client_id, secret=key, tenant=directory_id
-        )
-        self._dns_client = DnsManagementClient(credentials, sub_id)
+        # Store necessary initialization params
+        self._dns_client_handle = None
+        self._dns_client_client_id = client_id
+        self._dns_client_key = key
+        self._dns_client_directory_id = directory_id
+        self._dns_client_subscription_id = sub_id
+        self._dns_client = None
```

Comment:
Since the original property is `_dns_client` we'll need to double `_` here to avoid collision.

```suggestion
        self.__dns_client = None
```